### ISSUE TITLE
clang_delta remove-unused-function infinite recursion

### ISSUE THREAD

**Issue by user202729** [csmith-project/creduce#207]
(version: latest git master)

The file that crash the the program is rather large, but it can be obtained by preprocessing any file with `#include "range/v3/all.hpp"`.
 

The infinite recursion happens in `lookupFunctionDeclShallow`. With this patch applied, the crash doesn't happen: (however the program might be incorrect)

```diff
--- a/clang_delta/RemoveUnusedFunction.cpp
+++ b/clang_delta/RemoveUnusedFunction.cpp
@@ -706,7 +706,7 @@ RemoveUnusedFunction::lookupFunctionDeclShallow(const DeclarationName &DName,
   for (auto *I : Ctx->using_directives()) {
     const NamespaceDecl *ND = I->getNominatedNamespace();
     // avoid infinite recursion
-    if (ND->getLookupParent() == Ctx)
+    if (ND == Ctx || ND->getLookupParent() == Ctx)
       return NULL;
     if (const FunctionDecl *FD = lookupFunctionDeclShallow(DName, ND))
       return FD;
```

**Comment by marxin** [comment]
Reduced test-case:

```c++
namespace a {
inline namespace b {
using namespace a;
}
struct c;
namespace {
using a::c;
}
} // namespace a
```

**Comment by chenyang78** [comment]
@marxin Thank you so much!

**Comment by chenyang78** [comment]
I committed the fix in d83cbda. Thanks, @user202729 and @marxin !
